src/realm/obj.cpp

```diff
@@ -482,6 +419,87 @@ Mixed Obj::get_primary_key() const
         return Mixed{get_key()};
     }
 }
+template <class T>
+inline int Obj::cmp(const Obj& other, ColKey::Idx col_ndx) const
+{
+    T val1 = _get<T>(col_ndx);
+    T val2 = other._get<T>(col_ndx);
+    if (val1 < val2) {
+        return -1;
+    }
+    else if (val1 > val2) {
+        return 1;
+    }
+    return 0;
+}
+
+template <>
+inline int Obj::cmp<StringData>(const Obj& other, ColKey::Idx col_ndx) const
+{
+    StringData a = _get<StringData>(col_ndx);
+    StringData b = other._get<StringData>(col_ndx);
+
+    if (a.is_null() && !b.is_null())
+        return -1;
+    else if (b.is_null() && !a.is_null())
```

Comment:
nit: There is an inconsistent mix of styles in this function of using needless "else" or not :-)
Same thing in surrounding functions.

Reply:
Isn't it more that all the "else" aren't needed, due to the return statements?

Reply:
exactly.